src/analysis/souffle/authorization_logic.dl

```diff
@@ -28,75 +28,77 @@
 
 // Some party asserts the claim that the label of an access path should
 // remove the referenced tag from the set.
-.decl saysRemoveTag(speaker: Principal, path: AccessPath, tag: Tag)
+.decl saysRemoveTag(speaker: Principal, path: AccessPath, owner: Principal, tag: Tag)
```

Comment:
If we represent this with an ADT, we have to add support for ADTs to authorization logic, which might be a nontrivial feature while we are still in a prototyping phase.

Reply:
@bgogul Your comments are showing up in a weird order and sometimes appearing twice, so I'm not sure if you will have seen my other comment in this thread. So I just wanted to point out again that if we implement this using an ADT, I think it means we would need to support ADTs in the authorization logic frontend as well (you'd need to be able to write `somePrin says removeTag(path, {owner, tag})` or whatever the syntax would be). I think this is a nontrivial feature that we might be able to avoid for a while.

Reply:
Sounds good, we can defer use of ADTs to later.